Wrap parse failures in a map & leave parses failures alone

cap10morgan opened the pull request:
Re-fixes test.chuck test suite

cap10morgan committed:
Add Transit handlers for Failure in the client

borkdude commented on src/pod/babashka/instaparse.clj:
```diff
@@ -49,8 +49,20 @@
+(babashka.pods/add-transit-write-handler!
+  #{Failure}
+  \"%s\"
+  (fn [^Failure f] [(.index f) (.reason f)]))
```
shouldn't these be just keyword accesses? `(:index f) (:reason f)`?

borkdude commented:
CI fails

cap10morgan commented:
> CI fails

🤦🏻  Because now it's the wrong class on the client side. Whack-a-mole! Hopefully there's a way to make both test suites happy here...

borkdude commented:
Do you mean with test suites the JVM tests vs the native tests?

cap10morgan commented:
> Do you mean with test suites the JVM tests vs the native tests?

Yes those and the `test.chuck` tests. When I fix one the other(s) seem to break. Fun! 🤪

borkdude commented:
I noticed the JVM doesn't test anything, it's just an empty script :)

How does this pod relate to the test.chuck tests, does that use instaparse?

For local testing you could check out instaparse-bb and change this line either to run a JVM or a binary:

https://github.com/babashka/instaparse-bb/blob/7f6291cf0856cbe8bbe8ba6a166c61e5b16bc0f8/src/instaparse/core.bb#L6

borkdude commented:
Ah yes, test.chuck has a direct dependency on instaparse, I see

cap10morgan commented:
> I noticed the JVM doesn't test anything, it's just an empty script :)

@borkdude Do you want me to add `clojure -M test.clj` to that script?

cap10morgan committed:
Wrap parse failures in a map w/ qualified key

…and look for the key in our `failure?` implementation.

Don’t wrap parses failures because they are indicated by nil first elements in the returned collection.

cap10morgan commented:
OK https://github.com/babashka/pod-babashka-instaparse/pull/11/commits/4dc7d3078c34c0eb6ef258efd8c9f1989a8c27ae seems to make everyone happy. `instaparse.core/parse` failures are wrapped in a `{::failure result}` map and `instaparse.core/parses` failures are returned unmodified because they indicate failure by returning a collection whose first element is `nil` (at least according to test.chuck's tests).
